front-end/src/components/MainPage.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react'
+import BaseButton from "./BaseButton.jsx";
+import PostRequestButton from "./PostRequestButton.jsx"
+import RequestButton from "./RequestButton.jsx"
+import HelloWorldButton from "./HelloWorldButton.jsx";
+import VoiceFlowButton from "./VoiceFlowButton.jsx";
+import TranscriptUploadModal from "./TranscriptUploadModal.jsx";
+
+class MainPage extends Component {
+    state = {
+        showTranscriptUploadModal: false
+    }
+
+    toggleTranscriptUploadModal() {
+        this.setState({showTranscriptUploadModal: !this.state.showTranscriptUploadModal})
+        console.log("Set showTranscriptUploadModal to " + this.state.showTranscriptUploadModal)
+    }
+
+    render() {
+        return (
+            <div className="MainPage bg-purple-300 flex">
+              <div className="flex gap-y-10 w-full flex-col">
+                <div className="justify-center flex">
+                  <BaseButton click={() => console.log("Hello world")} text={"Button 1"} />
+                </div>
+                <div className="justify-center flex" data-testid="upload-transcript-button">
+                  <BaseButton click={() => this.toggleTranscriptUploadModal()} text={"Upload Transcript"} />
```

Comment:
@WeissSa Can you clarify this and maybe see how it runs on your computer? I must've committed these changes on Friday without re-checking the page because the button doesn't open the modal when coded like this.

Reply:
I fixed this in my recent PR. The function itself needed to be an arrow function, not the call in click. 
See https://github.com/BestFloww/BestFlow/pull/17#pullrequestreview-1143344008